pkg/cvo/availableupdates.go

```diff
@@ -82,30 +97,44 @@ func (optr *Operator) syncAvailableUpdates(ctx context.Context, config *configv1
 				}
 			}
 		}
-	}
 
-	if optr.injectClusterIdIntoPromQL {
-		conditionalUpdates = injectClusterIdIntoConditionalUpdates(clusterId, conditionalUpdates)
-	}
+		if optr.injectClusterIdIntoPromQL {
+			conditionalUpdates = injectClusterIdIntoConditionalUpdates(clusterId, conditionalUpdates)
+		}
 
-	if usedDefaultUpstream {
-		upstream = ""
-	}
+		if usedDefaultUpstream {
+			upstream = ""
+		}
 
-	optrAvailableUpdates.Upstream = upstream
-	optrAvailableUpdates.Channel = channel
-	optrAvailableUpdates.Architecture = desiredArch
-	optrAvailableUpdates.Current = current
-	optrAvailableUpdates.Updates = updates
-	optrAvailableUpdates.ConditionalUpdates = conditionalUpdates
-	optrAvailableUpdates.ConditionRegistry = optr.conditionRegistry
-	optrAvailableUpdates.Condition = condition
+		optrAvailableUpdates.Upstream = upstream
+		optrAvailableUpdates.Channel = channel
+		optrAvailableUpdates.Architecture = desiredArch
+		optrAvailableUpdates.Current = current
+		optrAvailableUpdates.Updates = updates
+		optrAvailableUpdates.ConditionalUpdates = conditionalUpdates
+		optrAvailableUpdates.ConditionRegistry = optr.conditionRegistry
+		optrAvailableUpdates.Condition = condition
+	}
 
 	optrAvailableUpdates.evaluateConditionalUpdates(ctx)
+
+	queueKey := optr.queueKey()
+	for _, conditionalUpdate := range optrAvailableUpdates.ConditionalUpdates {
+		if recommended := meta.FindStatusCondition(conditionalUpdate.Conditions, "Recommended"); recommended == nil {
+			klog.Warningf("Requeue available-update evaluation, because %q lacks a Recommended condition", conditionalUpdate.Release.Version)
```

Comment:
IMO we do not need a log here. Mostly because I do not see the value addition when we are saying we are  evaluating the risks again because no recommended updates are available .

Reply:
in this `Warningf` case, there is no `Recommended` condition on the conditional update, so we aren't saying "we are evaluating the risks again" anywhere outside of this log line.  We will say that we are evaluating the risk again a few lines down in the `Infof` case, where we have a `Recommended`, but its status is neither `True` nor `False` (probably because it's `Unknown`, probably because its PromQL is client-side throttled).  I think the `Warningf` case is very important to know about, and the `Infof`  case is somewhat important to know about, and neither one should occur very often in healthy-cluster logs.  We'll get lots of requeues in logs where there are conditional update risks but local Thanos is broken, but I think that's ok, because we are not happy about that situation.